### PR TITLE
First Task for Process Mining by Tim Edelmann

### PR DESCRIPTION
**Pull request by TJEWH**
## Features
As given in task description, a kmeans implementation was added using sklearn, assuming kmeans must not implemented from scratch. Futhermore a visualization via mathplot was added for convience purposes.

#### Summary
- kmeans implemented
- visualization added

## Testing
By default, kmeans creates non deterministic results, therefore a deterministic centroid initialization was implemented for result testing. A different approach would have been to use statistical testing which would be overhead for this task in my eyes, since large amounts of implementation rely on well maintained and widely used [sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html).

#### Summary
- cluster amounts equals unique labels
- calculated results match expected results (for deterministic cases)

### CODE REVIEW

**Comment by gbrltv** [comment]
Hello! The idea for this task was to implement k-means from scratch since it is a pretty simple algorithm, pseudocode and references were given. Therefore, the use of external libraries such as scikit-learn was not allowed. Please update your implementation accordingly and leave the _dummy_ files that were there.
